modules/afamqp/afamqp.c

```diff
@@ -463,8 +488,28 @@ afamqp_dd_connect(AMQPDestDriver *self, gboolean reconnect)
       goto exception_amqp_dd_connect_failed_init;
     }
 
-  ret = amqp_login(self->conn, self->vhost, self->max_channel, self->frame_size, 0,
-                   AMQP_SASL_METHOD_PLAIN, self->user, self->password);
+  switch (self->auth_method)
```

Comment:
Could you extract this login code to a function of it's own right? This connect function started to become to large.